commands/enrich/enrich.go

```diff
@@ -0,0 +1,304 @@
+		return false, err
+	}
+	return xml.Unmarshal(content, &x) == nil, nil
+}
+
+func (enrichCmd *EnrichCommand) Run() (err error) {
+	defer func() {
+		if err != nil {
+			var e *exec.ExitError
+			if errors.As(err, &e) {
+				if e.ExitCode() != coreutils.ExitCodeVulnerableBuild.Code {
+					err = errors.New("Scan command failed. " + err.Error())
+				}
+			}
+		}
+	}()
+	_, xrayVersion, err := xrutils.CreateXrayServiceManagerAndGetVersion(enrichCmd.serverDetails)
+	if err != nil {
+		return err
+	}
+
+	// Validate Xray minimum version for graph scan command
+	err = clientutils.ValidateMinimumVersion(clientutils.Xray, xrayVersion, scangraph.GraphScanMinXrayVersion)
+	if err != nil {
+		return err
+	}
+
+	log.Info("JFrog Xray version is:", xrayVersion)
+
+	threads := 1
+	if enrichCmd.threads > 1 {
+		threads = enrichCmd.threads
+	}
+
+	// resultsArr is a two-dimensional array. Each array in it contains a list of ScanResponses that were requested and collected by a specific thread.
+	resultsArr := make([][]*ScanInfo, threads)
+	fileProducerConsumer := parallel.NewRunner(enrichCmd.threads, 20000, false)
+	fileProducerErrors := make([][]formats.SimpleJsonError, threads)
+	indexedFileProducerConsumer := parallel.NewRunner(enrichCmd.threads, 20000, false)
+	indexedFileProducerErrors := make([][]formats.SimpleJsonError, threads)
+	fileCollectingErrorsQueue := clientutils.NewErrorsQueue(1)
+	// Start walking on the filesystem to "produce" files that match the given pattern
+	// while the consumer uses the indexer to index those files.
+	enrichCmd.prepareScanTasks(fileProducerConsumer, indexedFileProducerConsumer, resultsArr, indexedFileProducerErrors, fileCollectingErrorsQueue, xrayVersion)
+	enrichCmd.performScanTasks(fileProducerConsumer, indexedFileProducerConsumer)
+
+	// Handle results
+	flatResults := []xrutils.ScaScanResult{}
+	for _, arr := range resultsArr {
+		for _, res := range arr {
+			flatResults = append(flatResults, xrutils.ScaScanResult{Target: res.Target, XrayResults: []services.ScanResponse{*res.Result}})
+		}
+	}
+	if enrichCmd.progress != nil {
+		if err = enrichCmd.progress.Quit(); err != nil {
+			return err
+		}
+
+	}
+
+	fileCollectingErr := fileCollectingErrorsQueue.GetError()
+	var scanErrors []formats.SimpleJsonError
+	if fileCollectingErr != nil {
+		scanErrors = append(scanErrors, formats.SimpleJsonError{ErrorMessage: fileCollectingErr.Error()})
+	}
+	scanErrors = appendErrorSlice(scanErrors, fileProducerErrors)
+	scanErrors = appendErrorSlice(scanErrors, indexedFileProducerErrors)
+
+	scanResults := xrutils.NewAuditResults()
+	scanResults.XrayVersion = xrayVersion
+	scanResults.ScaResults = flatResults
+
+	isxml, err := isXML(scanResults.ScaResults)
+	if err != nil {
+		return
+	}
+	if isxml {
+		if err = xrutils.NewResultsWriter(scanResults).
+			AppendVulnsToXML(); err != nil {
+			return
+		}
+	} else {
+		if err = xrutils.NewResultsWriter(scanResults).
+			AppendVulnsToJson(); err != nil {
+			return
+		}
+	}
```

Comment:
```suggestion
    outFormat := formats.Json
    if isXml {
        outFormat = formats.Xml
    }
    if err = xrutils.NewResultsWriter(scanResults).
		SetOutputFormat(outFormat).
		SetIsMultipleRootProject(true).
		SetScanType(services.Binary).
		PrintScanResults(); err != nil {
		return
	}
```
If you are using the `results` you already have infrastructure to print Json or more formats, 
I would prefer that you will just add a new output format `XML` and implement it noramly so everyone using the `results` will benefit from it.
If you want to different output don't use the infrastructure.

Reply:
The output is a bit different. I'm printing the input + additional field and not the result from API so I cannot use it.

Reply:
if you are not using the ResultWriter infra, remove the methods from it.